spec/aerospike/util/connection_pool_spec.rb

```diff
@@ -49,6 +51,17 @@
       end
     end
 
+    context "enforce max connections as a hard limit" do
```

Comment:
This test is good for the expected behavior, but does not consider what happens when a few connections are closed. Will the client allow new connections to be created afterwards, or will be stuck at max_conns_exceeded forever? (Which from the code, I think it does)